tests/test_esx.py

```diff
@@ -165,6 +165,41 @@ def test_getHostGuestMapping(self, mock_client):
         result = self.esx.getHostGuestMapping()['hypervisors'][0]
         self.assertEqual(expected_result.toDict(), result.toDict())
 
+    @patch('suds.client.Client')
+    def test_getHostGuestMappingNoHostName(self, mock_client):
+        expected_hypervisorId = 'Fake_uuid'
+        expected_guestId = 'guest1UUID'
+        expected_guest_state = Guest.STATE_RUNNING
+
+        fake_parent = MagicMock()
+        fake_parent.value = 'fake_parent_id'
+        fake_parent._type = 'ClusterComputeResource'
+
+        fake_vm_id = MagicMock()
+        fake_vm_id.value = 'guest1'
+
+        fake_vm = MagicMock()
+        fake_vm.ManagedObjectReference = [fake_vm_id]
+        fake_vms = {'guest1': {'runtime.powerState': 'poweredOn',
+                               'config.uuid': expected_guestId}}
+        self.esx.vms = fake_vms
+
```

Comment:
Nitpick: I would add here note that: `'config.network.dnsConfig.hostName'` is not listed in `fake_host`.

Reply:
That is the point of the test. The hypervisor does not get reported on when the hostname is absent.

Reply:
Yeah, I know. :-) It would be nice to have such information in comment in source code, but comment here is enough too. :-)